File: 3394-check-if-grid-can-be-cut-into-sections/3394-check-if-grid-can-be-cut-into-sections.java
class Solution {
    public boolean checkValidCuts(int n, int[][] rectangles) {
        int m = rectangles.length;
        Arrays.sort(rectangles,(o1,o2)->Integer.compare(o1[0],o2[0]));
        int max = rectangles[0][2];
        int cnt = 0;
        for(int i=1;i<m;i++) {
            if(rectangles[i][0]>=max) cnt++;
            max = Math.max(max,rectangles[i][2]);
        }
        if(cnt>=2) return true;
        Arrays.sort(rectangles,(o1,o2)->Integer.compare(o1[1],o2[1]));
        max = rectangles[0][3];
        cnt = 0;
        for(int i=1;i<m;i++) {
            if(rectangles[i][1]>=max) cnt++;
            max = Math.max(max,rectangles[i][3]);
        }
        return cnt>=2;
    }
}

File: 3394-check-if-grid-can-be-cut-into-sections/3394-check-if-grid-can-be-cut-into-sections.py
from typing import List
class Solution:
    def checkValidCuts(self, n: int, rectangles: List[List[int]]) -> bool:
        xactions = []
        yactions = []
        for a, b, c, d in rectangles:
            xactions.append((a, 1))
            xactions.append((c, -1))
            yactions.append((b, 1))
            yactions.append((d, -1))
        xactions.sort()
        yactions.sort()
        tmp = 0
        cnt = 0
        for pos, value in xactions:
            tmp += value
            if tmp == 0:
                cnt += 1
        if cnt >= 3:
            return True
        tmp = 0
        cnt = 0
        for pos, value in yactions:
            tmp += value
            if tmp == 0:
                cnt += 1
        if cnt >= 3:
            return True
        return False